core/src/com/unciv/ui/overviewscreen/ReligionOverviewTable.kt

```diff
@@ -0,0 +1,117 @@
+import com.unciv.models.translations.tr
+import com.unciv.ui.utils.*
+import kotlin.math.min
+
+class ReligionOverviewTable(
+    private val viewingPlayer: CivilizationInfo,
+    private val overviewScreen: EmpireOverviewScreen
+): Table() {
+    
+    val gameInfo = viewingPlayer.gameInfo
+    private val topButtons = Table(CameraStageBaseScreen.skin)
+    private val topButtonLabel = "Click an icon to see the stats of this religion".tr().toLabel()
+    private val statsTable = Table(CameraStageBaseScreen.skin)
+    private val beliefsTable = Table(CameraStageBaseScreen.skin)
+    private var selectedReligion: String? = null
+    
+    init {
+        draw(null)
+    }
+    
+    private fun draw(religion: Religion?) {
+        topButtons.clear()
+        topButtonLabel.clear()
+        statsTable.clear()
+        beliefsTable.clear()
+        clear()
+
+        addReligionButtons()
+        if (religion != null) loadReligion(religion)
+        add(topButtons).pad(5f).row()
+        add(topButtonLabel).pad(5f)
+        addSeparator()
+        add(statsTable).pad(5f).row()
+        add(beliefsTable).pad(5f)
+    }
+    
+    private fun addReligionButtons() {
+        topButtons.clear()
+        val existingReligions: List<Religion> = gameInfo.civilizations.mapNotNull { it.religionManager.religion }
+        for (religion in existingReligions) {
+            val button: Button
+            if (religion.isPantheon()) {
+                val image = if (viewingPlayer.knows(religion.foundingCivName) || viewingPlayer.civName == religion.foundingCivName)
+                    ImageGetter.getNationIndicator(gameInfo.getCivilization(religion.foundingCivName).nation, 60f)
+                else
+                    ImageGetter.getRandomNationIndicator(60f)
+                button = Button(image, CameraStageBaseScreen.skin)
+            } else {
+                val image = ImageGetter.getReligionIcon(religion.iconName)
+                image.color = Color.BLACK
+                val icon = image.surroundWithCircle(60f)
+                button = Button(icon, CameraStageBaseScreen.skin)
+            }
+            
+            button.onClick {
+                selectedReligion = religion.name
+                draw(religion)
+            }
+            
+            if (selectedReligion == religion.name)
+                button.disable()
+            
+            topButtons.add(button).pad(5f)
+        }
+    }
+    
+    private fun loadReligion(religion: Religion) {
+        topButtonLabel.setText(religion.name.tr())
+        for (belief in religion.getPantheonBeliefs()) {
+            beliefsTable.add(createBeliefDescription(belief)).pad(10f).row()
+        }
+        for (belief in religion.getFollowerBeliefs()) {
+            beliefsTable.add(createBeliefDescription(belief)).pad(10f).row()
+        }
+        for (belief in religion.getFounderBeliefs()) {
+            beliefsTable.add(createBeliefDescription(belief)).pad(10f).row()
+        }
```

Comment:
Sounds like `for(belief in religion.getPantheonBeliefs() + ... + ...)`

Reply:
+ being efficient because they're sequences, some other Collection.plus() do copies.